concordium-consensus/src/Concordium/GlobalState/TransactionTable.hs

```diff
@@ -353,3 +356,52 @@ reversePTT trs ptt0 = foldr reverse1 ptt0 trs
                 upd (Just (low, high)) =
                         assert (low == sn + 1) $
                         Just (low - 1, high)
+
+-- * Transaction verification cache
+
+-- |The transaction verification cache stores transaction `CacheableVerificationResult`s associated with `TransactionHash`s.
+-- New entries are being put into the cache when receiving new transasactions (either as a single transaction or within a block)
+-- by `doReceiveTransaction` and `doReceiveTransactionInternal`. 
+-- The cached verification results are used by the Scheduler to short-cut verification
+-- during block execution.
+-- Entries in the cache are being expunged when the associated transaction is either
+-- finalized or purged.
+-- A note on `CacheableVerificationResult`s. These verification results are characterized by being verifiable in the future.
+-- That is, they are either deemed valid or they are possible valid in the future. (when received by one of the two functions mentioned above)
+
+type TransactionVerificationCache = HM.HashMap TransactionHash CacheableVerificationResult
+
+data CacheableVerificationResult
+  = VerificationResultSuccess
+  -- ^The transaction was valid.
+  deriving (Eq, Show)
```

Comment:
Because invalid transactions are immediately being rejected in the `doReceiveTransaction` and `doReceiveTransactionInternal` functions. 
Subsequent transactions (i.e. the same transactions) will be _filtered away_ by the deduplication in the node, thus there is no need to cache _rejected_ transactions, also it would require a _collision_ for them to actually be useful i.e., before we would make use of the cache. 

.... So I thought it would be better to just keep a _minimal_ cache. 

Would you prefer caching invalid ones as well?

Reply:
Also `VerificationResultSuccess` should probably be renamed something along the lines of 
`CredentialDeploymentVerificationResultSuccess`. 

The verification of chain updates will yield another "success" result, but one which contains a hash of the keys used to sign the `chainupdate`. Hence the `Scheduler` will need to check that the authorization keys have not changed.